### DISCUSSION TITLE
WPF

### DISCUSSION THREAD

**Discussion by Ruttie2006** [unoplatform/uno#10150] [Q&A] [answered]
I recently added the Skia.WPF platform to my project, but I get these errors:
```txt
fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get any remote control server endpoint from the IDE.
fail: Microsoft.UI.Xaml.Controls.CommandBarFlyout[0]
      The member CommandBarFlyout.CommandBarFlyout() is not implemented in Uno.
fail: Microsoft.UI.Xaml.Controls.TextCommandBarFlyout[0]
      The member TextCommandBarFlyout.TextCommandBarFlyout() is not implemented in Uno.
fail: Uno.UI.DataBinding.BindingPropertyHelper[0]
      The [ScrollViewer] property getter does not exist on type [Microsoft.UI.Xaml.Controls.TextBox]
fail: Microsoft.UI.Xaml.BrushTransition[0]
      The member BrushTransition.BrushTransition() is not implemented in Uno.
fail: Microsoft.UI.Xaml.BrushTransition[0]
      The member TimeSpan BrushTransition.Duration is not implemented in Uno.
fail: Microsoft.UI.Xaml.Controls.ContentPresenter[0]
      The member BrushTransition ContentPresenter.BackgroundTransition is not implemented in Uno.
fail: Microsoft.UI.Xaml.Controls.ScrollContentPresenter[0]
      The member float ZoomFactor is not implemented in Uno.
warn: Microsoft.UI.Xaml.UIElement[0]
      The BorderThicknessProperty dependency property does not exist on Microsoft.UI.Xaml.Controls.FontIcon
```
Any particular reason this happens? Am I using a component that does not exist on WPF? (If so, what component?)

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report, these are expected errors at this time, mostly because Uno uses the official styles from Microsoft which use more features than what Uno currently implements. We're working towards implementing those missing features as we go.

You can [use this property](https://github.com/unoplatform/uno/blob/1f825ea3bf047d667c9c7e31050c5c22e30a254f/src/Uno.UI/FeatureConfiguration.cs#L40) to adjust the logging of these failures.

**Reply by Ruttie2006** [reply]
Does this mean I should not be worried about this error?
If this means I'm using something that is not implemented in uno, do those errors reveal which/what type of component may be breaking?

**Reply by Ruttie2006** [reply]
Also, the property you referenced does not exist. It exists in the Uno.UI package, but when I install that it complains about Uno.WinUI.
Should Uno.WinUI only be used on the .Windows platform? Or is it fine as is?

**Reply by jeromelaban** [reply]
In general, no you can ignore those messages, that is unless the app does not do what you think it should do.

Note that this property is only available for Uno-only targets you need to use it with `#if HAS_UNO`.

**Reply by Ruttie2006** [reply]
I see, thank you.
